### ISSUE TITLE
CSP Meta extension error

### ISSUE THREAD

**Issue by MikeKovarik** [systemjs/systemjs#944]
I'm building app with Aurelia beta 1 using systemjs csp version 0.19.6 running in WinRT (Windows 10 app). And I don't really know what's happening since I'm just trying to build an app but whenever I export function, number, string or object from some of my modules (I'm using babel to translate my es6 code and modules) it breaks the app by (I have no idea how this is related) being unable to load templates which Aurelia loads using the !text plugin and throws error with this in the stack trace.

```
TypeError: Object doesn't support property or method 'match'
  at Anonymous function (/jspm_packages/system-csp-production.src.js:3676:7)
  at Anonymous function (/jspm_packages/system-csp-production.src.js:341:7)
```

Basically it only works when i export instance of Class otherwise it ends up with this.
From what I've seen in the code the line 3637 `var meta = load.source.match(metaRegEx);` expects `load.source` to be string but there ends up to be promise in my case. And this line is in the section called Meta Extension hence the name.
Other than that I don't know anything else. It's just a big wtf to me.

Update: I downgraded back to 0.18.17 which works fine

**Comment by guybedford** [comment]
@KenRmk note that the `!text` plugin is not designed to be compatible with System CSP as it requires fetching the text with XHR.

**Comment by KenRmk** [comment]
Yeah. I forgot to mention tat I managed to get around that by bundling all `!text` imports. It's a bit uncomfortable but works. Unfortunately there's still the problem I described in the first post. I'm sorry I couldn't describe it it grater detail. Though I'm currently using older version of SystemJS and it works.

**Comment by guybedford** [comment]
@KenRmk thanks for the update. Is there anyway at all you could share a repo or steps to get a reproduction of this issue? `load.source` should certainly not be a promise, which sounds like it could be a bug. Assistance tracking it down to fix it would be really welcome.
